### PR TITLE
Use `ops.rsqrt`, improve normalization layers and enable ops fusion in tflite

### LINKED ISSUE #824: tflite cannot fuse `BatchNormalization` in Keras Core as effectively as in the original Keras
Let's take MobileNetV3Small as example

```python
import tensorflow as tf
from keras_core.applications.mobilenet_v3 import MobileNetV3Small  # keras-core
from tensorflow import keras  # keras

keras_core_model = MobileNetV3Small(
    input_shape=(224, 224, 3), minimalistic=True
)
keras_model = keras.applications.MobileNetV3Small(
    input_shape=(224, 224, 3), minimalistic=True
)

for model, name in [
    (keras_core_model, "keras_core"),
    (keras_model, "keras"),
]:
    tf_callable = tf.function(
        model.call,
        input_signature=[tf.TensorSpec((1, 224, 224, 3), tf.float32)],
        autograph=True,
        jit_compile=True,
    )
    tf_concrete_function = tf_callable.get_concrete_function()
    converter = tf.lite.TFLiteConverter.from_concrete_functions(
        [tf_concrete_function], tf_callable
    )
    # do quantization...
    converter.optimizations = [tf.lite.Optimize.DEFAULT]
    tflite_model = converter.convert()
    with open(f"{name}_model.tflite", "wb") as f:
        f.write(tflite_model)
```

The visualization from netron:
Keras Core vs. Keras:
 

We can see that `BatchNormalization` was divided into `[Sub, Mul, Mul, Add]` compared to the fused op in the original Keras.

I believe the reason is that `BatchNormalization` in Keras Core is built using `ops`.
However, in the original Keras, `BatchNormalization` can switch to using `_fused_batch_norm` with `tf.compat.v1.nn.fused_batch_norm`

https://github.com/keras-team/keras/blob/94a1712027366cd626df4f1d77f25f918a18f456/keras/layers/normalization/batch_normalization.py#L978-L988

The unfused ops will reduce the inference speed. This issue makes Keras Core less suitable for tflite deployment.

### PR DESCRIPTION
**Pull request by james77777778**
Fixes #824 

This PR accomplishes the following:
1. adding support for `rsqrt` in numpy backend (using jax's impl)
2. replacing `1 / ops.sqrt(x)` with `ops.rsqrt` for improved speed
3. reordering the ops in normalization layers to unify the implementation and match the expression of `tf.nn.batch_normalization` [link](https://github.com/tensorflow/tensorflow/blob/1cb1a030a62b169d90d34c747ab9b09f332bf905/tensorflow/python/ops/nn_impl.py#L1533)
4. Ensuring 100% unit test coverage for all normalization layers

After completing 3, tflite recognizes the pattern of CONV+BN+ReLU, and the ops are fused successfully.

<details>
<summary> standalone MobileNetV3 export script </summary>

```python
import tensorflow as tf

from keras_core.applications.mobilenet_v3 import MobileNetV3Small

keras_core_model = MobileNetV3Small(
    input_shape=(224, 224, 3), minimalistic=True
)

tf_callable = tf.function(
    keras_core_model.call,
    input_signature=[tf.TensorSpec((1, 224, 224, 3), tf.float32)],
    autograph=True,
    jit_compile=True,
)
tf_concrete_function = tf_callable.get_concrete_function()
converter = tf.lite.TFLiteConverter.from_concrete_functions(
    [tf_concrete_function], tf_callable
)
converter.optimizations = [tf.lite.Optimize.DEFAULT]
tflite_model = converter.convert()
with open("model.tflite", "wb") as f:
    f.write(tflite_model)

```

</details>

The visualization from netron: (before this PR vs. after this PR)
 圖片

<details>
<summary> benchmark script </summary>

```python
from keras_core import layers
from keras_core import mixed_precision
from keras_core import models
from keras_core import ops

# "float32"
# "mixed_float16"
# "mixed_bfloat16"
dtype_policy = "float32"
mixed_precision.set_dtype_policy(dtype_policy)

x_train = ops.random.uniform(shape=(512, 64, 64, 64))
y_train = ops.random.uniform(shape=(512, 64, 64, 64))

# layers.BatchNormalization
# layers.GroupNormalization
# layers.LayerNormalization
normalization_cls = layers.LayerNormalization
normalization_args = {}
if normalization_cls is layers.GroupNormalization:
    normalization_args = {"groups": -1}

model = models.Sequential(
    [
        layers.InputLayer(shape=(64, 64, 64)),
        normalization_cls(**normalization_args),
        normalization_cls(**normalization_args),
        normalization_cls(**normalization_args),
    ]
)
model.compile(loss="mse", optimizer="adam")
model.fit(x_train, y_train, batch_size=128, epochs=3)

```

</details>

And the improvement:
|backend|layer|before this PR|after this PR|
|-|-|-|-|
|tensorflow|BatchNormalization|48ms/step|46ms/step|
|jax|BatchNormalization|49ms/step|48ms/step|
|torch|BatchNormalization|127ms/step|127ms/step|
|tensorflow|GroupNormalization|50ms/step|49ms/step|
|jax|GroupNormalization|51ms/step|50ms/step|
|torch|GroupNormalization|129ms/step|129ms/step|
|tensorflow|LayerNormalization|54ms/step|53ms/step|
|jax|LayerNormalization|55ms/step|54ms/step|
|torch|LayerNormalization|165ms/step|122ms/step|

### CODE REVIEW

File: keras_core/backend/numpy/math.py [hunk 1]
```diff
@@ -298,3 +299,7 @@ def istft(
     else:
         end = expected_output_len
     return x[..., start:end]
+
+
+def rsqrt(x):
+    return np.array(jax_rsqrt(x))
```
Comment (fchollet) [inline keras_core/backend/numpy/math.py:305]: Why not `1. / sqrt(x)`? It's numpy native, and we're not worried about performance for the numpy backend.

File: keras_core/layers/normalization/batch_normalization.py [hunk 2]
```diff
@@ -224,17 +224,24 @@ def call(self, inputs, training=None, mask=None):
...
+        res = -mean * inv
         if self.center:
             beta = ops.reshape(self.beta, broadcast_shape)
-            beta = ops.cast(beta, outputs.dtype)
-            outputs = outputs + beta
+            beta = ops.cast(beta, inputs.dtype)
+            res = res + beta
+
+        # Note: Folding BatchNormalization depends on the precise order of ops
+        # that are generated by the expression below
```
Comment (fchollet) [inline keras_core/layers/normalization/batch_normalization.py:243]: Good comment!

**Review by fchollet — COMMENTED** [review]
Thank you for the PR!

File: keras_core/backend/numpy/math.py (hunk 1 above)
Comment (james77777778) [reply keras_core/backend/numpy/math.py]: I was thinking of being consistent with other backends, but it should be okay to use `1. / sqrt(x)`
Fixed.

**Review by fchollet — APPROVED** [review]
LGTM -- thank you for the great contribution!
